## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 515090`
Commit: [`779db3e7a6085c13310be733fef4fbeee0e19d27`](https://github.com/NixOS/nixpkgs/commit/779db3e7a6085c13310be733fef4fbeee0e19d27) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/779db3e7a6085c13310be733fef4fbeee0e19d27..pull/515090/head))
Merge: [`7c69c65ce676e87765210cf5278fb5fc235a5069`](https://github.com/NixOS/nixpkgs/commit/7c69c65ce676e87765210cf5278fb5fc235a5069)

Logs: https://github.com/ToasterUwU/nixpkgs-review-gha/actions/runs/25175801181


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>cyberia</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>cyberia</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>cyberia</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>cyberia</summary>
<pre>27 | |     })
28 | |     .await
   | |__________^ cannot infer type

error[E0282]: type annotations needed
  --&gt; src/security/secrets.rs:34:5
   |
34 | /     tokio::task::spawn_blocking(move || {
35 | |         let entry = keyring::Entry::new(SERVICE, &amp;key)
36 | |             .map_err(|e| AppError::Storage(format!(&quot;Keyring entry error: {}&quot;, e)))?;
37 | |         entry
...  |
40 | |     })
41 | |     .await
   | |__________^ cannot infer type

Some errors have detailed explanations: E0282, E0433.
For more information about an error, try `rustc --explain E0282`.
error: could not compile `cyberia` (lib) due to 6 previous errors
       Error failed to build app: failed to build app</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>cyberia</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>cyberia</summary>
<pre>27 | |     })
28 | |     .await
   | |__________^ cannot infer type

error[E0282]: type annotations needed
  --&gt; src/security/secrets.rs:34:5
   |
34 | /     tokio::task::spawn_blocking(move || {
35 | |         let entry = keyring::Entry::new(SERVICE, &amp;key)
36 | |             .map_err(|e| AppError::Storage(format!(&quot;Keyring entry error: {}&quot;, e)))?;
37 | |         entry
...  |
40 | |     })
41 | |     .await
   | |__________^ cannot infer type

Some errors have detailed explanations: E0282, E0433.
For more information about an error, try `rustc --explain E0282`.
error: could not compile `cyberia` (lib) due to 6 previous errors
       Error failed to build app: failed to build app</pre>
</details>
</details>

--- a/pkgs/by-name/cy/cyberia/package.nix
+++ b/pkgs/by-name/cy/cyberia/package.nix
@@ -1,0 +1,69 @@
+{
+  rustPlatform,
+  fetchgit,
+  fetchNpmDeps,
+  cargo-tauri,
+  nodejs,
+  npmHooks,
+  pkg-config,
+  lib,
+  stdenv,
+  wrapGAppsHook4,
+  dbus,
+  glib-networking,
+  gtk3,
+  webkitgtk_4_1,
+  alsa-lib,
+}:
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "cyberia";
+  version = "0.2.3";
+
+  __structuredAttrs = true;
+
+  src = fetchgit {
+    url = "https://git.gay/zutyosh/Cyberia.git";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-REcp7rl3nDzgqaBBjbC+LN+WMKLdfyn3nn9Jvypo7Ps=";
+  };
+
+  cargoRoot = "src-tauri";
+  buildAndTestSubdir = finalAttrs.cargoRoot;
+  cargoHash = "sha256-FrYDjydBsW1nYc3xsiR5jaPDm4eRwoOb1DK6XOMyiLU=";
+
+  npmDeps = fetchNpmDeps {
+    name = "${finalAttrs.pname}-${finalAttrs.version}-npm-deps";
+    inherit (finalAttrs) src;
+    hash = "sha256-BUoR/+5gARwZndh7YYp7vA/InWkn30yF9kR6PXsITBo=";
+  };
+
+  nativeBuildInputs = [
+    cargo-tauri.hook
+
+    nodejs
+    npmHooks.npmConfigHook
+
+    pkg-config
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isLinux [ wrapGAppsHook4 ];
+
+  buildInputs = [
+    dbus
+    gtk3
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isLinux [
+    glib-networking
+    webkitgtk_4_1
+    alsa-lib
+  ];
+
+  meta = {
+    description = "VRCX style client for Resonite";
+    homepage = "https://git.gay/zutyosh/Cyberia";
+    changelog = "https://git.gay/zutyosh/Cyberia/releases/tag/v${finalAttrs.version}";
+    mainProgram = "cyberia";
+    platforms = lib.platforms.linux;
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ toasteruwu ];
+  };
+})

